### DISCUSSION TITLE
Can a teleport agent de-register itself?

### DISCUSSION THREAD

**Discussion by hendrikmaus** [gravitational/teleport#15345] [Q&A]
I noticed that stopping the teleport agent on a node doesn't remove it from the auth cluster right away. Instead it takes a couple of minutes for the node to disappear. This popped up for me when replacing instances and I suddenly saw all of them duplicated in the cluster for some minutes.

Is there an idiomatic way to tell the teleport cluster, from the agents perspective, that it is going down and it should be removed from the cluster instantly?

**Comment by webvictim** [comment]
There is not currently any way to do this, but I think it's a decent idea. Maybe raise a feature request? https://github.com/gravitational/teleport/issues/new?assignees=&labels=feature-request&template=feature_request.md
